test/index.spec.ts

```diff
@@ -0,0 +1,465 @@
+import axios from "axios";
+import { describe, it, expect, beforeEach } from "@jest/globals";
+
+import { UniversalProvider } from "@walletconnect/universal-provider";
+import { TezosToolkit } from "@taquito/taquito";
+import {
+  PartialTezosOperation as PartialTezosOperationOriginal,
+  PartialTezosOriginationOperation as PartialTezosOriginationOperationOriginal,
+  TezosDrainDelegateOperation,
+  TezosOperationType,
+} from "@airgap/beacon-types";
+
+import TezosProvider, { TezosProviderOpts } from "../src/TezosProvider";
+import {
+  TezosConnectionError,
+  TezosConnectOpts,
+  TezosInitializationError,
+  TezosMethod,
+} from "../src/types";
+import { TezosChainDataMainnet, UnsupportedOperations } from "../src/constants";
+import { SAMPLES, SAMPLE_KINDS } from "./samples";
+import { ScriptedContracts } from "@taquito/rpc";
+
+interface PartialTezosOriginationOperation
+  extends Omit<PartialTezosOriginationOperationOriginal, "script"> {
+  script: ScriptedContracts;
+}
+
+type PartialTezosOperation =
+  | Exclude<PartialTezosOperationOriginal, PartialTezosOriginationOperationOriginal>
+  | PartialTezosOriginationOperation;
+
+jest.mock("@walletconnect/universal-provider");
+jest.mock("@taquito/taquito");
+jest.mock("axios");
+
+describe("TezosProvider", () => {
+  let provider: TezosProvider;
+  const mockInit = jest.fn();
+  const mockConnect = jest.fn();
+  const mockRequest = jest.fn();
+
+  beforeEach(() => {
+    (UniversalProvider.init as jest.Mock).mockResolvedValue({
```

Comment:
it's better to use jest.mocked(UniversalProvider.init) because it respects types